pyproject.toml

```diff
@@ -68,7 +68,7 @@ protobuf = "^3.0.0"
 
 altair = {version = "^4.2.2", optional = true}
 tensorflow-macos = { version = "~2.12", optional = true}
-streamlit = {version = "^1.22.0", optional = true}
+streamlit = "1.28.2"
```

Comment:
What's the reason for setting streamlit to the fixed version `1.28.2`.  I would recommend to use `^1.28.2` or `~1.28.2`. What about the optional flag for streamlit. I assume that this is still necessary.